packages/react-components/src/components/RTTModal.tsx

```diff
@@ -0,0 +1,108 @@
+import { _preventDismissOnEvent } from '@internal/acs-ui-common';
+/* @conditional-compile-remove(rtt) */
+import {
+  buttonsContainerClassName,
+  buttonStyles,
+  themedCaptionsSettingsModalStyle,
+  titleClassName,
+  titleContainerClassName
+} from './styles/CaptionsSettingsModal.styles';
+/* @conditional-compile-remove(rtt) */
+import { useLocale } from '../localization';
+
+/* @conditional-compile-remove(rtt) */
+/**
+ * @beta
+ * strings for rtt modal
+ */
+export interface RTTModalStrings {
+  rttModalTitle?: string;
+  rttModalText?: string;
+  rttConfirmButtonLabel?: string;
+  rttCancelButtonLabel?: string;
+  rttModalAriaLabel?: string;
+  rttCloseModalButtonAriaLabel?: string;
+}
+/* @conditional-compile-remove(rtt) */
+/**
+ * @beta
+ * RTTModal Component Props.
+ */
+export interface RTTModalProps {
+  strings?: RTTModalStrings;
+  showModal?: boolean;
+  onDismissModal?: () => void;
+  onStartRTT?: () => Promise<void>;
+}
+/* @conditional-compile-remove(rtt) */
+/**
+ * @beta
+ * a component for rtt modal
+ */
+export const RTTModal = (props: RTTModalProps): JSX.Element => {
+  const { showModal, onDismissModal, onStartRTT } = props;
+  const localeStrings = useLocale().strings.rttModal;
+  const strings = { ...localeStrings, ...props.strings };
+
+  const theme = useTheme();
+
+  const onDismiss = useCallback((): void => {
+    if (onDismissModal) {
+      onDismissModal();
+    }
+  }, [onDismissModal]);
+
+  const onConfirm = useCallback(async (): Promise<void> => {
+    if (onStartRTT) {
+      await onStartRTT();
+    }
+    onDismiss();
+  }, [onDismiss, onStartRTT]);
+
+  const RTTModalStyle: Partial<IModalStyles> = useMemo(() => themedCaptionsSettingsModalStyle(theme), [theme]);
+
+  return (
+    <>
+      {
+        <Modal
+          titleAriaId={strings?.rttModalAriaLabel}
+          isOpen={showModal}
+          onDismiss={onDismiss}
+          isBlocking={true}
+          styles={RTTModalStyle}
+        >
+          <Stack horizontal horizontalAlign="space-between" verticalAlign="center" className={titleContainerClassName}>
+            <Text className={titleClassName}>{strings?.rttModalTitle}</Text>
+            <IconButton
+              iconProps={{ iconName: 'Cancel' }}
+              ariaLabel={strings?.rttCloseModalButtonAriaLabel}
+              onClick={onDismiss}
+              style={{ color: theme.palette.black }}
+            />
+          </Stack>
+          <Text>{strings?.rttModalText}</Text>
+
+          <Stack horizontal horizontalAlign="end" className={buttonsContainerClassName}>
+            <PrimaryButton styles={buttonStyles(theme)} onClick={onConfirm}>
+              <span>{strings?.rttConfirmButtonLabel}</span>
+            </PrimaryButton>
+            <DefaultButton onClick={onDismiss} styles={buttonStyles(theme)}>
+              <span>{strings?.rttCancelButtonLabel}</span>
+            </DefaultButton>
+          </Stack>
+        </Modal>
+      }
+    </>
+  );
+};
```

Comment:
This looks very similar to our Prompt composite component here https://github.com/Azure/communication-ui-library/blob/3f044549c3d930521ff02f1bc563957793b355d4/packages/react-composites/src/composites/CallComposite/components/Prompt.tsx. 
I wonder if you can use that component instead like how it is used in this PR where a prompt is created to confirm soft muting all participants.
https://github.com/Azure/communication-ui-library/pull/4734/files#diff-9f558acc72c8ad424e14522e01b71ab2b48bc892571f4d8992c71e43314be4a0R328.
This prompt is also used for spotlight and end call for everyone.
Is this only going to be used only in composite?

Reply:
I notice that for the prompt component it takes onConfirm?: () => void;, however for RTT modal it needs to call an async function when confirmed to start RTT. 

We also need the styling for this modal to be consistent with captions modal. And we need it to be public so contoso can use this modal to start RTT. 

Do we have a storybook page for this prompt component? And can we make prompt public?

Reply:
I thought this component will be internal in storybook so Contoso wouldn't be able to import it. Prompt is not exposed to Contoso so I thought we could keep it consistent with the confirmation modals.
 
You can also change the type of onConfirm, onCancel, and onClose to be async of Prompt since it will not break for sync functions. Prompt also takes in styles in it's prop.

But I will approve for now to unblock you. You can always refactor in the future unless it will be exposed in the future.